src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
Yes, I know the contract. But we're talking about the way to provide a custom identity strategy, not custom equals/hashCode implementations. equals/hashCode is just one implementation of that strategy. And in our case, hashCode is not even necessary for the implementation.